domain/src/main/java/com/anytypeio/anytype/domain/multiplayer/SpaceViewSubscriptionContainer.kt

```diff
@@ -205,4 +205,19 @@ fun SpaceViewSubscriptionContainer.isSharingLimitReached(
         val isLimitReached = limit == 0 || count >= limit
         Pair(isLimitReached, limit)
     }
-}
\ No newline at end of file
+}
+
+fun SpaceViewSubscriptionContainer.sharedSpaceCount(
+    spaceToUserPermissions: Flow<Map<Id, SpaceMemberPermissions>>
+) : Flow<Int> {
+    return combine(
+        observe(),
+        spaceToUserPermissions
+    ) { spaceViews, permissions ->
+        spaceViews.count { spaceView ->
```

Comment:
please add Timber.d("Total shared space count: $count") and .catch {  Timber.e(exception, "Error calculating shared space count") }

Reply:
done